pulumi-eks-go/main.go

```diff
@@ -0,0 +1,439 @@
+		            "Service": "eks.amazonaws.com"
+		        },
+		        "Action": "sts:AssumeRole"
+		    }]
+		}`),
+		})
+		if err != nil {
+			return err
+		}
+		eksPolicies := []string{
+			"arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+			"arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+		}
+		for i, eksPolicy := range eksPolicies {
+			_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf(prefix+"rpa-%d", i), &iam.RolePolicyAttachmentArgs{
+				PolicyArn: pulumi.String(eksPolicy),
+				Role:      eksRole.Name,
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		// Resource: Security Groups
+		// Purpose: A security group acts as a virtual firewall, controlling the traffic that is allowed to reach and leave the resources that it is associated with.
+		// Docs: https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html
+
+		// Create a Security Group that we can use to actually connect to our cluster
+		clusterSg, err := ec2.NewSecurityGroup(ctx, prefix+"-cluster-sg", &ec2.SecurityGroupArgs{
+			VpcId: vpc.ID(),
+			Egress: ec2.SecurityGroupEgressArray{
+				ec2.SecurityGroupEgressArgs{
+					Protocol:   pulumi.String("-1"),
+					FromPort:   pulumi.Int(0),
+					ToPort:     pulumi.Int(0),
+					CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+				},
+			},
+			Ingress: ec2.SecurityGroupIngressArray{
+				ec2.SecurityGroupIngressArgs{
+					Protocol:   pulumi.String("tcp"),
+					FromPort:   pulumi.Int(80),
+					ToPort:     pulumi.Int(80),
+					CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+				},
+			},
+			Tags: pulumi.StringMap{
+				"Name": pulumi.String(prefix + "-cluster-sg"),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Resource: EKS Cluster
+		// Purpose: The Amazon EKS control plane & Amazon EKS nodes that are registered with the control plane
+		// Docs: https://docs.aws.amazon.com/eks/latest/userguide/clusters.html
+
+		// EKS cluster
+		var subnetIds pulumi.StringArray
+
+		cluster, err := eks.NewCluster(ctx, prefix+"-cluster", &eks.ClusterArgs{
+			RoleArn: pulumi.StringOutput(eksRole.Arn),
+			VpcConfig: &eks.ClusterVpcConfigArgs{
+				PublicAccessCidrs: pulumi.StringArray{
+					pulumi.String("0.0.0.0/0"),
+				},
+				SecurityGroupIds: pulumi.StringArray{
+					clusterSg.ID().ToStringOutput(),
+				},
+				SubnetIds: pulumi.StringArray(
+					append(
+						subnetIds,
+						privSubnet1.ID(),
+						privSubnet2.ID(),
+						privSubnet3.ID(),
+						pubSubnet1.ID(),
+						pubSubnet2.ID(),
+						pubSubnet3.ID(),
+					)),
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Resource: IAM Role
+		// Purpose: An IAM role is an IAM identity that you can create in your account that has specific permissions.
+		// Docs: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
+
+		// Create the EC2 NodeGroup Role
+		nodeGroupRole, err := iam.NewRole(ctx, prefix+"nodegroup-iam-role", &iam.RoleArgs{
+			AssumeRolePolicy: pulumi.String(`{
```

Comment:
I think it's useful to move these policies elsewhere since it's a string possibly a constant? However, it's only used once and the other examples of this on the interweb appear to have it inline.

Reply:
Ah, figured out a way
```go
		resourceTags := make(map[string]string)

		resourceTags["CreatedBy"] = "pulumi-eks-go"
		resourceTags["GitOrg"] = "gsweene2"
		resourceTags["GitRepo"] = "pulumi"
```
and usage
```go
			Tags:               pulumi.ToStringMap(resourceTags),
```

Reply:
https://github.com/gsweene2/pulumi/pull/1/commits/716b247c4698914bd82ba60f3b947f3488af4bed